### PR TITLE
JDK-6690019: JOptionPane obscured behind an always on top JFrame hangs UI

### PR DESCRIPTION
**Pull request by prsadhuk**
A JOptionPane message dialog owned by a JFrame which is not always on top can be obscured behind an `"always on top" JFrame`. Since the option pane is modal, one can't move the "always on top frame" to reveal it, so the application effectively hangs as we cannot dismiss the option pane dialog
Fix is to make sure the option pane dialog is always on top....

---------
### Progress
- [ ] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-6690019](https://bugs.openjdk.org/browse/JDK-6690019): JOptionPane obscured behind an always on top JFrame hangs UI (**Bug** - P4)

### Reviewers
 * [Tejesh R](https://openjdk.org/census#tr) (@TejeshR13 - Committer)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/16060/head:pull/16060` \
`$ git checkout pull/16060`

Update a local copy of the PR: \
`$ git checkout pull/16060` \
`$ git pull https://git.openjdk.org/jdk.git pull/16060/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 16060`

View PR using the GUI difftool: \
`$ git pr show -t 16060`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/16060.diff">https://git.openjdk.org/jdk/pull/16060.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/16060#issuecomment-1749302080)

### CODE REVIEW

**Comment by TejeshR13** [comment]
The fix is working fine with the test mentioned in JBS. Any reason for not adding a test for the fix?

**Review by TejeshR13 — COMMENTED** [review]
The fix is working fine with the test mentioned in JBS. Any reason for not adding a test for the fix?

**Comment by prsadhuk** [comment]
> The fix is working fine with the test mentioned in JBS. Any reason for not adding a test for the fix?

I couldn't get ways to do automated test for this and manual test using PassFailJFrame doesn't support multiple test windows which this test needs..

**Comment by alisenchung** [comment]
Would it be better to check if the JFrame is "always on top" before making the JOptionPane also always on top?

**Comment by prsadhuk** [comment]
> Would it be better to check if the JFrame is "always on top" before making the JOptionPane also always on top?

No..That's the issue this PR is trying to solve...
If a frame is not "always on top" and is hidden by a frame which is "always on top", then the JOptionPane owned by former frame will hang the process as it is modal and will not allow events to go to latter or former frame..

**Review by TejeshR13 — APPROVED** [review]

**Comment by prrace** [comment]
This seems to me like an application bug.
Apart from a couple of special cases - Popup being one of them - we don't use this call internally in dialogs etc.
As much as possible determining the behaviours of these windows should be up to the app.
I think this should be closed as not a (JDK) bug.

File: test/jdk/javax/swing/JOptionPane/TestJOptionPaneAlwaysOnTopBug.java [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+                    alwaysOnTopFrame.dispose();
+                }
+                if (normalFrame != null) {
+                    normalFrame.dispose();
+                }
+            });
+        }
+    }
+
+    private static void centreFrame(JFrame myFrame)  {
```
Comment (turbanoff) [inline test/jdk/javax/swing/JOptionPane/TestJOptionPaneAlwaysOnTopBug.java:95]:
```suggestion
    private static void centreFrame(JFrame myFrame) {
```

**Comment by prsadhuk** [comment]
> This seems to me like an application bug. Apart from a couple of special cases - Popup being one of them - we don't use this call internally in dialogs etc. As much as possible determining the behaviours of these windows should be up to the app. I think this should be closed as not a (JDK) bug.

I get your point but can we consider this as a "defensive" fix much like DiD fix that we do, in case application does't take care of it, so that we dont get this kind of hang situation...We have print dialog as always-on-top so we can consider this optionpane modal dialog also be the same...
If any issues arise, we can always roll it back later on..

**Comment by prrace** [comment]
I think its more likely to cause bugs than fix them. As I said there are special cases and the Print Dialog is one of them because the app can't easily get a handle to it.

**Pull request closed without merging**